File: java/src/com/alofmethbin/jadawin/elements/Heading.java
package com.alofmethbin.jadawin.elements;

import com.alofmethbin.jadawin.Article;
import java.util.List;

public class Heading extends Element {
    private String text;

    public Heading( Article article, List<String> lines) {
        super( article);
        text  = lines.get(0);
    }

    @Override
    public boolean isMultiline() {
        return false;
    }
}

File: ruby/elements/heading.rb
require_relative 'image'

module Elements
  class Heading < Element
    include Utils

    def initialize( compiler, article, text)
      super( article)
      @text = check_label( article, text[0])
    end

    def multiline?
      false
    end

    def to_data( compiler, article)
      {'type'  => 'heading',
       'text'  => @text}
    end
  end
 end